### PR TITLE
Micrometer performance - use Meter.MeterProvider

### LINKED ISSUE #40597: Micrometer performance improvements - Stork and binder
### Description

Implement improvements described in #40507 on the `io.quarkus.micrometer.runtime.binder.stork`and `io.quarkus.micrometer.runtime.binder` packages.
The identified classes are:

- `StorkObservationCollectorBean` 
- `RestClientMetricsFilter`

### PR DESCRIPTION
**Pull request by mcruzdev**
Fixes #40597

### CODE REVIEW

**Review by ebullient — APPROVED** [review]

**Comment by brunobat** [comment]
Still missing the enhancements on `RestClientMetricsFilter` and the PR is still on Draft.

**Comment by mcruzdev** [comment]
Thank you for the feedback @brunobat, I will fix test and handle the `RestClientMetricsFilter`

**Comment by brunobat** [comment]
can you please squash your commits, @mcruzdev ?

File: extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/binder/stork/StorkObservationCollectorBean.java [hunk 1]
```diff
@@ -21,59 +22,66 @@
...
+        this.instanceCounter.withTags(tags).increment(observation.getDiscoveredInstancesCount());
+        this.serviceDiscoveryTimer.withTags(tags).record(observation.getServiceDiscoveryDuration().getNano(),
+                TimeUnit.NANOSECONDS);
+        this.serviceSelectionTimer.withTags(tags).record(observation.getServiceSelectionDuration().getNano(),
+                TimeUnit.NANOSECONDS);
 
         if (observation.failure() != null) {
             if (observation.isServiceDiscoverySuccessful()) {
-                serviceSelectionFailures.increment();
+                this.serviceSelectionFailures.withTags(tags).increment();
```
Comment (brunobat) [inline extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/binder/stork/StorkObservationCollectorBean.java:82]: There were no tags before... But probably it's better to have them.

Comment (mcruzdev) [reply extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/binder/stork/StorkObservationCollectorBean.java:82]: Yes, I miss the tags in the first commit.

**Review by brunobat — APPROVED** [review]

File: extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/binder/stork/StorkObservationCollectorBean.java (hunk 1 above)
Comment (brunobat) [reply extensions/micrometer/runtime/src/main/java/io/quarkus/micrometer/runtime/binder/stork/StorkObservationCollectorBean.java:82]: When I say before I mean, before the PR... CC @aureamunoz

**Comment by mcruzdev** [comment]
Solving ...

**Review by brunobat — APPROVED** [review]

**Comment by brunobat** [comment]
Thanks @mcruzdev !

**Pull request merged**